File: v1/Demo/source/demo_get_pos_limited.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
    usleep(1000000); 
 
    vector<float> pos_limited = api.get_pos_limited();
    if (pos_limited.size() != 12)
    {
        Logger::get_instance()->print_trace_error("get pos_limited failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get pos_limited success : ";
    for (int i = 0; i < pos_limited.size(); ++i) {
        std::cout << pos_limited[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}

File: v1/Demo/demo_get_pos_limited.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    pos_limited = api.get_pos_limited()

    if len(pos_limited) != 12:
        logger.print_trace_error("get pos_limited failed")
    else:
        logger.print_trace("get pos_limited success", pos_limited)

if __name__ == "__main__":
    main()


